crates/iceberg-catalog/src/catalog/tables.rs

```diff
@@ -492,9 +504,12 @@ impl<C: Catalog, A: AuthZHandler, S: SecretStore>
             .storage_config
             .storage_profile
             .file_io(storage_secret.as_ref())?;
+        let compression_codec =
```

Comment:
let's move this to line 447, then we don't end up calling db and all other things if we're gonna reject the request due to unsupported compression anyways

Reply:
I moved it to 484 because at that point the "correct" "one and only" result is available.

Reply:
You're right, then let's not do that, the updates may , I also just saw that we are already extracting the codec in `implementations/postgres/tabular/table.rs:575` which kinda short circuits the function. To fully avoid doing unnecessary DB roundtrips, we'd have to check if `request.updates` has a `TableUpdate::SetProperties` where `updates` contains a unsupported codec.

Reply:
The properties of the request are put into `validate_table_properties`, however that function just checks if there is an unsupported property and if it is in lowercase. 

Might it be a good idea to extend that function to check for valid values for properties, where possible? 

... or should a `CommitTableRequest` already be valid at that point, at least as far as it is possible to check? It should be possible to implement that by `impl Deserialize for CreateTableRequest`, but I am not sure if it should be done like this.

I think implementing a solution for this might warrant its' own issue.

Reply:
sure, let's revisit at a later point